### PR TITLE
Packaging

### PR DESCRIPTION
**Pull request by marcorossi5**
Hi all

With the current PR, I would like to address an issue found with TensorFlow and `pip` package management.
The problem is that sometimes the user has already installed TensorFlow on a machine using a specific distribution optimized for the platform: e.g. tensorflow-gpu, tensorflow-rocm...
Then, if he does `pip install pdfflow` on the machine, `pip` does not resolve correctly the TensorFlow dependence and instead of saying `requirement already satisfied`, it triggers a new installation of the common `tensorflow` package.  After that is really likely that the user ends up with a messed environment.

The second idea is that according to [setuptools documentation](https://setuptools.pypa.io/en/latest/userguide/quickstart.html#transitioning-from-setup-py-to-setup-cfg), we should probably move from a dynamic `setup.py` to a static `setup.cfg`.

Just one critical point in the current version: the PyPi publish workflow.
https://github.com/N3PDF/pdfflow/blob/a10df025e0f3e8e803feff58136155e2742d2b30/.github/workflows/pythonpublish.yml#L16-L26
This should be changed, but I am not sure how to do it right now. I'll figure out.
Please, let me know if you have comments.
Thanks

### CODE REVIEW

**Comment by marcorossi5** [comment]
I tried to edit the [pythonpublish.yaml](https://github.com/N3PDF/pdfflow/blob/packaging/.github/workflows/pythonpublish.yml) file according to this [doc](https://packaging.python.org/en/latest/tutorials/packaging-projects/).

I hope this works, even if I am not sure how to test it w/o publishing a new release.

**Comment by scarlehoff** [comment]
You can try it by manually running each step of the workflow but uploading the packages to https://test.pypi.org/ instead.

**Comment by marcorossi5** [comment]
I've manually checked the GitHub Action in [pythonpublish.yaml](https://github.com/N3PDF/pdfflow/blob/packaging/.github/workflows/pythonpublish.yml) and it works up to `twine upload dist/*`.

If you are satisfied with the changes of this PR we can merge I guess.

File: README.md [hunk 1]
```diff
@@ -19,24 +18,42 @@ The documentation for PDFFlow can be consulted in the readthedocs page: [pdfflow
...
+of TensorFlow for his platform. As such, by default, `pip` will not check it as
+a requirement.
+
+However, the user can also install it specifying a `MODE` option in the
+`pip` command. The list below summarizes the valid choices for the `MODE` flag:
+
+- `tf`: installs the `tensorflow` package
+- `tf-cpu`: installs the `tensorflow-cpu` package
+- `tf-gpu`: installs the `tensorflow-gpu` package
+- `tf-arm`: installs the `tensorflow-rocm` package
```
Comment (scarlehoff) [inline README.md:49]:
```suggestion
- `tf-amd`: installs the `tensorflow-rocm` package
```

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    sphinx_rtd_theme
+    recommonmark
+    sphinxcontrib-bibtex
+examples =
+    matplotlib
+    vegasflow
+tf = tensorflow
+tf-cpu = tensorflow-cpu
+tf-gpu = tensorflow-gpu
+tf-arm = tensorflow-rocm
```
Comment (scarlehoff) [inline setup.cfg:48]:
```suggestion
tf-amd = tensorflow-rocm
```

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    numpy >= 1.21
+    pyyaml
+    lhapdf_management
+
+[options.packages.find]
+where = src
+
+[options.entry_points]
+console_scripts =
+    quake = quake.scripts.quake:main
```
Comment (scarlehoff) [inline setup.cfg:34]: ?

**Comment by marcorossi5** [comment]
Ops, I missed to remove this old code parts, now everything should be resolved

**Review by scarlehoff — APPROVED** [review]
Great, thanks!

**Comment by scarlehoff** [comment]
Sure. You mean on the files that you modified? (I'm always using `black` so if that's what you using as well feel free to do so, the effect should be the same)

**Comment by marcorossi5** [comment]
> You mean on the files that you modified?
Usually I do it on all the files that `black` is able to parse

> I'm always using black so if that's what you using as well feel free to do so, the effect should be the same
Perfect ! I'll do it on my own then. I just wanted to be sure to have homogenous results .

After that, I'll leave merging to you

**Pull request merged**